upgrade/upgrade-scripts/execute

```diff
@@ -69,92 +70,93 @@ EOF
 apt_get update || die "failed to update apt sources"
 
 #
-# First make sure delphix-platform-<platform> is installed. This package
-# is a dependency of delphix-entire-<platform> and would be installed
-# even if we just directly installed delphix-entire-<platform>. However,
-# there are problems related to virtual packages that arise if we just
-# directly install delphix-entire-<platform>. In particular, if
-# delphix-entire is installed first, the following can happen (using
-# AWS as the example for the platform on which the upgrade is being
-# done):
+# To enable the use of "autoremove" later in this script to remove all
+# packages that are no longer required after the upgrade, we need to
+# mark all packages currently installed as "auto" installed. This way,
+# after explicitly installing all packages for the new "delphix-entire"
+# version, the only "auto" installed packages still remaining on the
+# system will be ones that aren't required by the new "delphix-entire".
 #
-#  - We run 'apt-get install delphix-entire-aws'
-#
-#  - Apt sees delphix-entire-aws relies on delphix-virtualization, so it
-#    marks that package and its dependencies for installation. One of
-#    its dependencies is the virtual package delphix-platform.
+dpkg-query -Wf '${Package}\n' | xargs apt-mark auto
```

Comment:
I think we need to make an exception for `delphix-kernel-$(uname -r)`. This will insure that any kernel version besides the one running currently _should_ be removed by auto-remove.

Reply:
Intersting, I'll try that out.

In my testing last night, even the previous/current kernel was removed, so hopefully this will fix that issue.